src/test/java/org/broadinstitute/hellbender/utils/mcmc/PosteriorSummaryTest.java

```diff
@@ -0,0 +1,87 @@
+package org.broadinstitute.hellbender.utils.mcmc;
+
+import org.apache.commons.math3.distribution.BetaDistribution;
+import org.apache.commons.math3.distribution.NormalDistribution;
+import org.apache.commons.math3.random.RandomGenerator;
+import org.apache.commons.math3.random.RandomGeneratorFactory;
+import org.apache.spark.api.java.JavaSparkContext;
+import org.broadinstitute.hellbender.engine.spark.SparkContextFactory;
+import org.broadinstitute.hellbender.utils.test.BaseTest;
+import org.testng.Assert;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Random;
+import java.util.stream.Collectors;
+
+/**
+ * Tests for {@link PosteriorSummary}.  Tests that posterior mode and highest-posterior-density credible interval
+ * are recovered for samples drawn from various posterior distributions.
+ *
+ * @author Samuel Lee &lt;slee@broadinstitute.org&gt;
+ */
+public final class PosteriorSummaryTest extends BaseTest {
+    private static final int RANDOM_SEED = 42;
+    private static final RandomGenerator rng = RandomGeneratorFactory.createRandomGenerator(new Random(RANDOM_SEED));
+
+    private static final List<Double> normalSamples = toList(new NormalDistribution(rng, 10., 1).sample(100));
+    private static final List<Double> normalSamplesSmall = toList(new NormalDistribution(rng, 10., 1).sample(10));
+    private static final List<Double> normalSamplesScaled = toList(new NormalDistribution(rng, 10., 0.01).sample(100));
+
+    private static final List<Double> betaSamples = toList(new BetaDistribution(rng, 2, 10).sample(100000));
+    private static final List<Double> betaSamplesEdgeMode = toList(new BetaDistribution(rng, 10, 1).sample(1000));
+
```

Comment:
I would like to see one more test case (which ought to be very similar to the true posterior on MAFs) in which you take a Beta centered near 1/2 (so that it has large weight above 1/2) and reflect it so that f > 1/2 is mapped to 1 - f.